include/fastgltf/types.hpp

```diff
@@ -1170,31 +1183,67 @@ namespace fastgltf {
 		}
 
 		template <typename F>
-		[[nodiscard]] OptionalWithFlagValue<T> and_then(F&& func)& {
+		[[nodiscard]] auto and_then(F&& func)& {
+			using U = std::remove_cv_t<std::remove_reference_t<std::invoke_result_t<F, T&>>>;
```

Comment:
[cppreference](https://en.cppreference.com/w/cpp/utility/optional/and_then) uses `std::remove_cvref_t`, so I opted for it as well. I'd guess they don't use `std::decay_t` because function return types don't need to be decayed.

Reply:
`std::remove_cvref` is a C++20 feature

Reply:
Fair point, this is fine then.